mypy/plugin.py

```diff
@@ -379,3 +392,189 @@ def int_pow_callback(ctx: MethodContext) -> Type:
         else:
             return ctx.api.named_generic_type('builtins.float', [])
     return ctx.default_return_type
+
+
+def add_method(
+        info: TypeInfo,
+        method_name: str,
+        args: List[Argument],
+        ret_type: Type,
+        self_type: Type,
+        function_type: Instance) -> None:
+    from mypy.semanal import set_callable_name
+
+    first = [Argument(Var('self'), self_type, None, ARG_POS)]
+    args = first + args
+
+    arg_types = [arg.type_annotation for arg in args]
+    arg_names = [arg.variable.name() for arg in args]
+    arg_kinds = [arg.kind for arg in args]
+    assert None not in arg_types
+    signature = CallableType(cast(List[Type], arg_types), arg_kinds, arg_names,
```

Comment:
Is it possible to avoid the ``cast``, why it is necessary? We try to avoid using casts.

Reply:
The named tuple code does the same thing.  arg_types is technically a `List[Optional[Type]]`  but note the `assert None not in arg_types` right above to make sure it's right.

Reply:
I think the cast is not required with that assert. Have you actually tried removing it?

Reply:
Yeah I get:

```
mypy/plugin.py:580: error: Argument 1 to "CallableType" has incompatible type "List[Optional[Type]]"; expected "List[Type]"
```

When running `mypy mypy/plugin.py --strict`